src/ServerlessPluginBespoken.ts

```diff
@@ -0,0 +1,407 @@
+ * 
+ * - `serverless deploy --enablePassThru`
+ *   When --enablePassThru is passed on command line, the plugin will intercept the serverless package lifecycle and replace the deployed bundle with passthru lambda handlers
+ *   that forward requests to the bespoken proxy server (and subsequently back to developer machine)
+ */
+export class ServerlessPluginBespoken {
+  private serverless: any;
+  private options: any;
+  private proxy: BSTProxy;
+  private originalServicePath: string;
+  private passThruServicePath: string;
+
+  private get selectedFunctionFromCommandLine(): string | undefined {
+    return this.options.function;
+  }
+
+  private get functionsFromServerlessConfig(): any {
+    return this.serverless.service.functions;
+  }
+
+  private get environmentVariablesFromServerlessConfig(): object {
+    const env = this.serverless.service.provider.environment;
+    if (env) {
+      this.serverless.cli.log(
+        "Configuring environment variables from serverless config"
+      );
+    } else {
+      this.serverless.cli.log(
+        "No Environment variables found in serverless config"
+      );
+    }
+    return env || {};
+  }
+
+  private get handlers(): string[] {
+    const handlers = this.serverless.service.functions || {};
+
+    return Object.entries(handlers).map(([_, lambdaDefinition]) => {
+      return lambdaDefinition.handler;
+    });
+  }
+
+  private get enableSecurity(): boolean {
+    return !(this.options.secure == null);
+  }
+
+  public commands = {
+    proxy: {
+      usage: "Plugin to call bespoken bst lambda service",
+      lifecycleEvents: ["start"],
+      options: {
+        function: {
+          usage:
+            "Specify the function in your config that you want to use" +
+            '(e.g. "--function myFunction" or "-f myFunction")',
+          required: false,
+          shortcut: "f"
+        },
+        secure: {
+          usage:
+            "Make proxy server require that 'secure' token be specified -- used to reduce likelihood of arbitrary hosts contacting your service via the proxy server",
+          required: false,
+          shortcut: "s"
+        }
+      }
+    }
+  };
+
+  public get hooks() {
+    return {
+      "proxy:start": this.proxyStart,
+      "before:package:createDeploymentArtifacts": this.deployPassThru,
+      "after:package:createDeploymentArtifacts": this.injectPassThruModules
+    };
+  }
+
+  constructor(serverless: any, options: any) {
+    this.serverless = serverless;
+    this.options = options;
+  }
+
+  proxyStart = async () => {
+    // parse the bespoken config
+    await Global.initializeCLI();
+    // enable verbose logging
+    LoggingHelper.setVerbose(true);
+
+    // ensure environment variables from serverless config are set
+    mutateProcessEnvironmentVariables(
+      this.environmentVariablesFromServerlessConfig
+    );
+
+    // create the lambda proxy
+    if (this.selectedFunctionFromCommandLine) {
```

Comment:
This validation should be:
if (!passthroughOption) {
  // single function mode -- assumes not secure proxy
      const handler = extractHandlerObject(
       this.functionsFromServerlessConfig,
        // This can be null in this point in which case the function is taken from the serverless config
        this.selectedFunctionFromCommandLine      );
} else {
  passthrough logic
}